File: khalilBranch/HuntKingdom 4.0/src/ProduitBundle/Entity/Promotion.php
<?php

namespace ProduitBundle\Entity;
use Symfony\Component\Validator\Constraints as Assert;
use Doctrine\ORM\Mapping as ORM;


/**
 * Promotion
 *
 * @ORM\Table(name="promotion")
 * @ORM\Entity(repositoryClass="ProduitBundle\Repository\PromotionRepository")
 */
class Promotion
{
    /**
     * @var int
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */

    private $id;
    /**
     * @var \DateTime
     *
     * @ORM\Column(name="dateFin", type="datetime", options={"default" :"CURRENT_TIMESTAMP"})
     */

    private $dateFin;
    /**
     * @var \DateTime
     *@Assert\GreaterThan("today")
     * @ORM\Column(name="dateDebut", type="datetime", options={"default" :"CURRENT_TIMESTAMP"})
     */


    private $dateDebut;
    /**
     * @var float
     * @Assert\GreaterThanOrEqual(0)
     * @ORM\Column(name="prix", type="float",options={"default" :0})
     */
    private $prix;
    /**
     * @var int
     * @Assert\GreaterThanOrEqual(0)
     * @ORM\Column(name="pourcentage", type="integer")
     */
    private $pourcentage;

    /**
     * @ORM\ManyToOne(targetEntity="Produit")
     * @ORM\JoinColumn(name="idProduit", referencedColumnName="id")
     */
    private $idProduit;

    /**
     * @var int
     *
     * @ORM\Column(name="active", type="integer",options={"default" :0})
     */
    private $active;



    /**
     * Get id
     *
     * @return int
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set pourcentage
     *
     * @param integer $pourcentage
     *
     * @return Promotion
     */
    public function setPourcentage($pourcentage)
    {
        $this->pourcentage = $pourcentage;

        return $this;
    }

    /**
     * Get pourcentage
     *
     * @return integer
     */
    public function getPourcentage()
    {
        return $this->pourcentage;
    }

    /**
     * Set idProduit
     *
     * @param \ProduitBundle\Entity\Produit $idProduit
     *
     * @return Promotion
     */
    public function setIdProduit(\ProduitBundle\Entity\Produit $idProduit = null)
    {
        $this->idProduit = $idProduit;

        return $this;
    }

    /**
     * Get idProduit
     *
     * @return \ProduitBundle\Entity\Produit
     */
    public function getIdProduit()
    {
        return $this->idProduit;
    }

    /**
     * Set dateFin
     *
     * @param \DateTime $dateFin
     *
     * @return Promotion
     */
    public function setDateFin($dateFin)
    {
        $this->dateFin = $dateFin;

        return $this;
    }

    /**
     * Get dateFin
     *
     * @return \DateTime
     */
    public function getDateFin()
    {
        return $this->dateFin;
    }

    /**
     * Set dateDebut
     *
     * @param \DateTime $dateDebut
     *
     * @return Promotion
     */
    public function setDateDebut($dateDebut)
    {
        $this->dateDebut = $dateDebut;

        return $this;
    }

    /**
     * Get dateDebut
     *
     * @return \DateTime
     */
    public function getDateDebut()
    {
        return $this->dateDebut;
    }

    /**
     * Set prix
     *
     * @param float $prix
     *
     * @return Promotion
     */
    public function setPrix($prix)
    {
        $this->prix = $prix;

        return $this;
    }

    /**
     * Get prix
     *
     * @return float
     */
    public function getPrix()
    {
        return $this->prix;
    }

    /**
     * Set active
     *
     * @param integer $active
     *
     * @return Promotion
     */
    public function setActive($active)
    {
        $this->active = $active;

        return $this;
    }

    /**
     * Get active
     *
     * @return integer
     */
    public function getActive()
    {
        return $this->active;
    }
}

File: SPRINT MOBILE/OLD STUFF/INTEGRATION/HK SIKR 0.2 Rafik is here/HuntKingdom/src/com/mycompany/entities/Promotion.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.entities;
import java.util.Date;
/**
 *
 * @author khalil
 */
public class Promotion {
     private int id ;
    private int pourcentage;
    private Produit Produit;
    private Date dateFin;
    private Date dateDebut;
    private float prix;
    private int active;

    public Promotion(int pourcentage, Produit Produit, Date dateFin, Date dateDebut,  int active) {
        this.pourcentage = pourcentage;
        this.Produit = Produit;
        this.dateFin = dateFin;
        this.dateDebut = dateDebut;
        this.active = active;
    }
  

    public Promotion(int id, int pourcentage, Produit Produit, Date dateFin, Date dateDebut) {
        this.id = id;
        this.pourcentage = pourcentage;
        this.Produit = Produit;
        this.dateFin = dateFin;
        this.dateDebut = dateDebut;
    }

    public Promotion(int id, int pourcentage, Date dateFin, Date dateDebut, int active,Produit p) {
        this.id = id;
        this.pourcentage = pourcentage;
        this.dateFin = dateFin;
        this.dateDebut = dateDebut;
        this.active = active;
    
        this.Produit=p;
    }

    public Promotion(int id, int pourcentage, Produit Produit, Date dateFin, Date dateDebut, float prix, int active) {
        this.id = id;
        this.pourcentage = pourcentage;
        this.Produit = Produit;
        this.dateFin = dateFin;
        this.dateDebut = dateDebut;
        this.prix = prix;
        this.active = active;
    }

    public Promotion(int id, int pourcentage, Produit Produit, Date dateFin, Date dateDebut, float prix) {
        this.id = id;
        this.pourcentage = pourcentage;
        this.Produit = Produit;
        this.dateFin = dateFin;
        this.dateDebut = dateDebut;
        this.prix = prix;
    }

    public Promotion() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getPourcentage() {
        return pourcentage;
    }

    public void setPourcentage(int pourcentage) {
        this.pourcentage = pourcentage;
    }

    public Produit getProduit() {
        return Produit;
    }

    public void setProduit(Produit Produit) {
        this.Produit = Produit;
    }

    public Date getDateFin() {
        return dateFin;
    }

    public void setDateFin(Date dateFin) {
        this.dateFin = dateFin;
    }

    public Date getDateDebut() {
        return dateDebut;
    }

    public void setDateDebut(Date dateDebut) {
        this.dateDebut = dateDebut;
    }

    public float getPrix() {
        return prix;
    }

    public void setPrix(float prix) {
        this.prix = prix;
    }

    public int getActive() {
        return active;
    }

    public void setActive(int active) {
        this.active = active;
    }

    @Override
    public String toString() {
        return "Promotion{" + "id=" + id + ", pourcentage=" + pourcentage + ", Produit=" + Produit + ", dateFin=" + dateFin + ", dateDebut=" + dateDebut + ", prix=" + prix + ", active=" + active + '}';
    }

    
}
